### PR TITLE
yarpcerrors: FromError.Unwrap expected result

### PR DESCRIPTION
**Pull request by jkanywhere**
Errors returned by `yarpcerrors.FromError` now behave as expected when
using `errors.Is` and `errors.Unwrap`.

`(*Status).Unwrap` implementation unwraps the inner error. Therefore,
always wrap the inner error when creating a `*yarpcerrors.Status` during
`FromError` to match the behavior of `Newf`.

Keeps the error constructed inside `Newf` invisible to consumers of
YARPC.

This diff also makes it safe to unwrap a nil `*Status`.

Fixes: T6597121

- [x] Description and context for reviewers: one partner, one stranger
- [ ] Docs (package doc)
- [x] Entry in CHANGELOG.md

### CODE REVIEW

**Comment by jkanywhere** [comment]
Supersedes https://github.com/yarpc/yarpc-go/pull/1966.

**Review by AlexAPB — APPROVED** [review]
Thanks! @twilly can I get your second pair of eyes before merging?

File: yarpcerrors/errors.go
Comment (twilly) [inline yarpcerrors/errors.go:73]: This is technically correct but not the most efficient. `fmt.Errorf` will format the string and then wrap the error. We could just wrap the error in the same way fmt does.

See: https://golang.org/src/fmt/errors.go?s=624:674#L7

Comment (AlexAPB) [reply yarpcerrors/errors.go:73]: sgtm, @jkanywhere would you mind adding a error wrapper type that does the same thing?

**Comment by AlexAPB** [comment]
@jkanywhere would you mind cleaning this up soon, I'm hoping to do a release tomorrow and would like to include this :)

**Comment by jkanywhere** [comment]
ack, will do

**Review by AlexAPB — APPROVED** [review]
Thanks!

**Pull request merged**
